### ISSUE TITLE
Improve Inference runtime.

### ISSUE THREAD

**Issue by amuamushu** [amuamushu/adv_avod_ssn#6]
Inference takes a long time to run even with small amounts of data. 

I can look into trying out the different suggestions made on this piazza post (https://piazza.com/class/ktx8mj1ysnd47o?cid=229)

**Things to Try**
- [x] Moving output directory to where code is. 
  - Currently, the output directory for checkpoints and results is located in the `teams` directory which has slow reads and writes. We did this because training takes a while so we didn't want to both train the same model.
- [ ] Moving code to `teams` directory.
- [x] Run on a subset of the data stored in the `home` directory

**Comment by amuamushu** [comment]
Our test target is set up such that it reads a small subset of the data and stores the outputs in both cases in the same location as the code. Even with only 3 samples, our test target is slow, meaning option 1 and 3 are likely slow with the whole dataset.

**Comment by amuamushu** [comment]
Ran a scenario where the output is where the code is but the data is in `teams`. Still the same speed.

**Comment by amuamushu** [comment]
> Ran a scenario where the output is where the code is but the data is in `teams`. Still the same speed.

As an update, I compared the log where the output directory was in `teams` (`simple_run_0107`) with the log where the output directory is in the code repo (`simple_run_0130`). The inference progress in both cases are similar, with `simple_run_0107` 700 steps ahead. In both cases, the resources were the same.

[simple_run_0107.txt](https://github.com/amuamushu/adv_avod_ssn/files/7973070/simple_run_0107.txt)
[simple_run_0130.txt](https://github.com/amuamushu/adv_avod_ssn/files/7973096/simple_run_0130.txt)

Another thing I could try is running image and lidar inference separately. While both can not finish in ~7 hours, maybe one of them can.

**Comment by amuamushu** [comment]
Talked to a TA about improving runtime of existing code. His suggestions are

1. make sure the model is not updating the weights --> pytorch: model.train() is off
2. input in the inference doesn't need gradient computations --> pytorch: input.requires_grad(False)
3. other direction: increase batch_size as much as memory fits

Make sure `nvidia-smi` shows percentage when it's running.

**Comment by amuamushu** [comment]
Added `device` config to bash scripts. More GPU seems to be used now. Before, only 8% of it was being used:
![image](https://user-images.githubusercontent.com/35519361/153023769-34664392-c4d2-4829-8f22-7269fe07214c.png)

Though, the GPU usage zeros out during the run often:
![image](https://user-images.githubusercontent.com/35519361/153024300-0150c244-9a4a-4abc-9c51-1dfc3ebc4a39.png)

**Comment by amuamushu** [comment]
won't do bc of time constraint
